### PR TITLE
prune linting on `qnode.py` and `execution.py`

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

Over time, the amount of linting rules we've silenced on `qnode.py` and `execution.py` has grown to be rather large.  In some cases, we are still silencing things that aren't even a problem anymore. In others, we are silencing complaints that have extremely easy fixes.

**Description of the Change:**

Prune's the list of things we are silencing in `qnode.py` and `execution.py` to either:

1) Fix the problem
2) Silence pylint as close to the problem as possible to avoid false negatives

**Benefits:**

Code more in line with pylint recommendations and fewer false negatives going forward.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/workflow/qnode.py [hunk 1]
```diff
@@ -1064,7 +1051,11 @@ def __call__(self, *args, **kwargs) -> qml.typing.Result:
         return self._impl_call(*args, **kwargs)
 
 
-qnode = lambda device, **kwargs: functools.partial(QNode, device=device, **kwargs)
+# pylint: disable=missing-docstring  set manually below
+def qnode(device, **kwargs):
```
Comment (dwierichs) [inline pennylane/workflow/qnode.py:1055]:
```suggestion
def qnode(device, **kwargs):
    """Docstring will be updated below."""
```

File: pennylane/workflow/execution.py [hunk 2]
```diff
@@ -16,11 +16,6 @@
 differentiation support.
 """
 
-# pylint: disable=import-outside-toplevel,too-many-branches,not-callable,unexpected-keyword-arg
-# pylint: disable=unused-argument,unnecessary-lambda-assignment,inconsistent-return-statements
-# pylint: disable=invalid-unary-operand-type,isinstance-second-argument-not-valid-type
-# pylint: disable=too-many-arguments,too-many-statements,function-redefined,too-many-function-args,too-many-positional-arguments
```
Comment (dwierichs) [inline pennylane/workflow/execution.py:22]: :exploding_head:

**Review by dwierichs — APPROVED** [review]
Nice cleanup work ensuring code quality! :tada:

Had a tiny non-critical suggestion.

File: pennylane/workflow/qnode.py (hunk 1 above)
Comment (albi3ro) [reply pennylane/workflow/qnode.py]: handled locally.

**Review by andrijapau — APPROVED** [review]
Awesome! 🥇

**Pull request merged**